public/classes/loom.ts

```diff
@@ -30,18 +31,31 @@ export class Loom {
 		}
 	}
 
-	async startWeaving(){
-		const userInput = this.getUserInput();
+    async startWeaving() {
+        const userInput = this.getUserInput();
+        const textile = new Textile(userInput);
+        const loomState: LoomState = {
+            numWarps: userInput.cols,
+            numRows: userInput.rows,
+            loomHeight: this.height,
+            loomWidth: this.width,
+            textile,
+            // TODO make this magic range code a util https://stackoverflow.com/a/37980601/14308614
+            warpState: [...Array(userInput.cols).keys()].map(_ => ({
+            position: WarpPosition.Under,
+            weftColor: 'blue', // This default should be taken from the list of thread colors
+            })),
+    	};
+		this.workingTextile = textile
 
 		drawLoomWeavingArea(this.height, this.width);
 
-		this.workingTextile = new Textile(userInput);
-
 		for(let i = 0; i <= userInput.rows; i++){
-			drawLoomState(userInput.cols, userInput.rows, i, this.workingTextile, this.height, this.width); //Should find a better way? we're passing a lot here
+			const loomCanvasInputs = drawLoomState(loomState, this.workingTextile, i);
```

Comment:
Your dev nose works well to detect this bad code smell. After, I bundled these parameters into a `LoomState` object, it made it really easy to change fields when developing the canvas code. Thanks for being our non-anosmic!